shell/platform/embedder/embedder_external_view_embedder.cc

```diff
@@ -0,0 +1,223 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/embedder/embedder_external_view_embedder.h"
+
+#include <algorithm>
+
+#include "flutter/shell/platform/embedder/embedder_render_target.h"
+
+namespace flutter {
+
+EmbedderExternalViewEmbedder::EmbedderExternalViewEmbedder(
+    RenderTargetCallback render_target_callback,
+    PresentCallback present_callback)
+    : render_target_callback_(render_target_callback),
+      present_callback_(present_callback) {
+  FML_DCHECK(render_target_callback_);
+  FML_DCHECK(present_callback_);
+}
+
+EmbedderExternalViewEmbedder::~EmbedderExternalViewEmbedder() = default;
+
+void EmbedderExternalViewEmbedder::Reset() {
+  pending_recorders_.clear();
+  pending_params_.clear();
+  composition_order_.clear();
+}
+
+// |ExternalViewEmbedder|
+bool EmbedderExternalViewEmbedder::HasPendingViewOperations() {
+  return false;
```

Comment:
This is semantically incorrect (e.g we may have "pending view operations"), I'd leave a comment explaining why we return false.

Reply:
Done.